packages/components/src/DataList/DataList.types.ts

```diff
@@ -226,10 +235,18 @@ export interface DataListActionProps<T extends DataListObject> {
   /**
    * The callback function when the action is clicked.
    */
-  readonly onClick?: (data: T) => void;
+  readonly onClick?: (data?: T) => void;
+}
+
+export interface DataListBulkActionProps
```

Comment:
I know this is being used in line 216, but that won't actually change the type of actions

<img width="1232" alt="image" src="https://github.com/GetJobber/atlantis/assets/15986172/80065d7e-6bbe-4309-ad7b-267cd1b35c9a">

still thinks a data might show up there.

One thing we could do tho is to have a `DataList.ItemAction` and `DataList.BatchAction` that both resolves back to just `DataListAction`.


```ts
DataList.ItemAction = DataListAction;
DataList.BatchAction = function DataListBatchAction(props: PropsWithoutThatDataInTheParam) => <DataListAction {...props} />
```